### PR TITLE
feat(components,typography): include type styles and update in components

### LINKED ISSUE #1546: Update V10 Experimental Components with correct type tokens from @carbon/type
Type tokens weren't finalized when we started building components. All experimental components will need to be updated.

Type tokens can be found here
https://next.carbondesignsystem.com/guidelines/typography/productive

### PR DESCRIPTION
**Pull request by joshblack**
Closes #1546 

#### Changelog

**New**

- Test for `_typography.scss`
- Add `type-style` mixin in code

**Changed**

- Update a variety of components to include `type-style` mixin

**Removed**

### CODE REVIEW

**Comment by joshblack** [comment]
Note to reviewers, still adding in tokens for some of the remaining ones. Will sadly not include all type styles as the list can be very hard to get right all in one go 😞 Will include a `feat` commit in the latest work I check-in 👍

**Review by asudoh — APPROVED** [review]
Thanks a lot @joshblack for looking through the components and making changes! LGTM 👍 from my perspective, given it involves UI changes to components, adding @IBM/carbon-designers as reviewers in case they want to take a look.

**Comment by asudoh** [comment]
BTW @joshblack could you please fix unused variable definitions reported in CI? Thanks!

**Review by tw15egan — APPROVED** [review]
Looks good 👍

File: src/components/modal/_modal.scss [hunk 1]
```diff
@@ -221,29 +221,30 @@
   }
 
   .#{$prefix}--modal-header__label {
-    @include typescale('omega');
-    @include letter-spacing;
+    @include type-style('label-01');
+
     color: $text-02;
     font-weight: 500;
```
Comment (aagonzales) [inline src/components/modal/_modal.scss]: remove `font-weight` label is rendering as bold still but should be like below:
image
 
Currently: 
image

File: src/components/notification/_inline-notification.scss [hunk 2]
```diff
@@ -181,14 +179,15 @@
...
-    font-weight: 600;
+    @include type-style('heading-01');
+
     margin: 0 $spacing-2xs 0 0;
     white-space: nowrap;
   }
 
   .#{$prefix}--inline-notification__subtitle {
-    @include typescale('zeta');
+    @include type-style('body-long-01');
```
Comment (aagonzales) [inline src/components/notification/_inline-notification.scss]: We changed this to `body-short-01` to help with information density.

File: src/components/notification/_toast-notification.scss [hunk 3]
```diff
@@ -181,21 +180,23 @@
...
-    @include letter-spacing;
+    @include type-style('heading-01');
+
     font-weight: 600;
     margin-top: 1rem;
   }
 
   .#{$prefix}--toast-notification__subtitle {
-    @include typescale('zeta');
+    @include type-style('body-long-01');
```
Comment (aagonzales) [inline src/components/notification/_toast-notification.scss]: We changed this to `body-short-01` to help with information density.

**Review by aagonzales — COMMENTED** [review]
Just small changes. Exciting to see the type tokens in there!

**Comment by joshblack** [comment]
Thanks @aagonzales! Updated 👍

**Review by aagonzales — APPROVED** [review]
Looks great! So exciting!!

**Pull request merged**